HW6/HW6.py

```diff
@@ -1,19 +1,542 @@
+
+    # 2 categoricals
+    if len(cat_predictors) > 1:
+        var1_list, var2_list, v_list = [], [], []
+        for x1 in cat_predictors:
+            for x2 in cat_predictors:
+                v = dafrenchy_correlation.cat_correlation(df[x1], df[x2])
+                var1_list.append(x1)
+                var2_list.append(x2)
+                v_list.append(v)
+
+        # create the table
+        d = {"Var1": var1_list, "Var2": var2_list, "CramersV": v_list}
+        cat_df = pd.DataFrame(data=d)
+        cat_df.sort_values(by="CramersV", ascending=False, inplace=True)
+        filename = "/results/" + "Cat_corr_table.html"
+        cat_df.to_html(filename, header="Categorical_variable_correlation_table")
+        print(path + filename)
+
+        # alternate heatmap
+        pivot = cat_df.pivot_table("CramersV", ["Var1"], "Var2")
+        fig = ff.create_annotated_heatmap(
+            z=pivot.to_numpy(), x=pivot.index.to_list(), y=pivot.index.to_list()
+        )
+        filename = "/results/" + "Categorical_correlation_matrix.html"
+        fig.write_html(file=filename, include_plotlyjs="cdn")
+        print(path + filename)
+
+    # cont/categorical
+
+    # categories first
+    if len(cont_predictors) > 0 and len(cat_predictors) > 0:
+        var1_list, var2_list, c_list = [], [], []
+        for x1 in cat_predictors:
+            for x2 in cont_predictors:
+                c = dafrenchy_correlation.cat_cont_correlation_ratio(df[x1], df[x2])
+                var1_list.append(x1)
+                var2_list.append(x2)
+                c_list.append(c)
+
+        # create the table
+        d = {"Var1": var1_list, "Var2": var2_list, "Correlation": c_list}
+        cat_cont_df = pd.DataFrame(data=d)
+        cat_cont_df.sort_values(by="Correlation", ascending=False, inplace=True)
+        filename = "/results/" + "Cat_cont_corr_table.html"
+        cat_cont_df.to_html(
+            filename, header="Continuous_Categorical_variable_correlation_table"
+        )
+        print(path + filename)
+
+        # alternate heatmap
+        pivot = cat_cont_df.pivot_table("Correlation", ["Var1"], "Var2")
+        fig = ff.create_annotated_heatmap(
+            z=pivot.to_numpy(), x=list(pivot.columns), y=pivot.index.to_list()
+        )
+        filename = "/results/" + "Continuous_vs_Categorical_correlation_matrix.html"
+        fig.write_html(file=filename, include_plotlyjs="cdn")
+        print(path + filename)
+
+    # sklearn
+    # break out the test and training based on the year
+    y_train = df[df["year"] != 2012][response]
+    y_test = df[df["year"] == 2012][response]
+    x_train = df[df["year"] != 2012][predictors]
+    x_test = df[df["year"] == 2012][predictors]
+
+    forest = RandomForestClassifier(random_state=4321)
+    forest.fit(x_train, y_train)
+    forest_prediction = forest.predict(x_test)
+    accuracy_array = forest_prediction == y_test
+    match = 0
+    total = 0
+    for item in accuracy_array:
```

Comment:
seems that you might have been able to use confusion matrix to calc this instead of doing it manually